### PR TITLE
textviewer finished

### PR DESCRIPTION
**Pull request by folax**

### CODE REVIEW

File: wgtfindreplacetext.cpp [hunk 1]
```diff
@@ -58,21 +62,100 @@ FindReplaceText::FindReplaceText(QWidget *parent)
...
+//replaceAll button slot
+void FindReplaceText::replaceAllSlot()
+{
+    flags = setOptions();
+    newWord = textReplaceEdit->text();
+    emit replaceAllSignal(textFindEdit->text(), flags, newWord);
+}
+
+//block buttons if text is empty
+void FindReplaceText::isEmptyText(QString text)
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.cpp:121]: const QString &

File: wgtfindreplacetext.cpp [hunk 2]
```diff
@@ -58,21 +62,100 @@ FindReplaceText::FindReplaceText(QWidget *parent)
...
+{
+    QList<QPushButton*> list = this->findChildren<QPushButton*>();
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.cpp:123]: for (QPushButton _btn : this->findChildren<QPushButton_>()) {
    btn->setEnabled(!text.isEmpty());
}

Code less, create more ;)

File: wgtfindreplacetext.cpp [hunk 3]
```diff
@@ -58,21 +62,100 @@ FindReplaceText::FindReplaceText(QWidget *parent)
...
+            list.at(i)->setEnabled(false);
+        }
+    } else {
+        for (int i = 0; i < list.size(); ++i) {
+            list.at(i)->setEnabled(true);
+        }
+    }
+}
+
+QTextDocument::FindFlags FindReplaceText::setOptions()
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.cpp:135]: Bad function name, use "QTextDocument::FindFlags FindReplaceText::options() const"

File: wgtfindreplacetext.h [hunk 4]
```diff
@@ -12,11 +13,26 @@ class QRadioButton;
 
 class FindReplaceText : public QDialog
 {
+    Q_OBJECT
 public:
     explicit FindReplaceText(QWidget *parent = 0);
+    void isReplace(bool);
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.h:19]: Missed "const"

File: wgtfindreplacetext.h [hunk 5]
```diff
@@ -12,11 +13,26 @@ class QRadioButton;
...
     ~FindReplaceText();
 
+public slots:
+    void findSlot();
+    void replaceSlot();
+    void replaceAllSlot();
+    void isEmptyText(QString);
+
+signals:
+    void findSignal(QString, QTextDocument::FindFlags);
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.h:29]: const QString &

File: wgtfindreplacetext.h [hunk 6]
```diff
@@ -12,11 +13,26 @@ class QRadioButton;
...
+    void replaceSignal(QString, QTextDocument::FindFlags, QString);
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.h:30]: See comment above

File: wgtfindreplacetext.h [hunk 7]
```diff
@@ -12,11 +13,26 @@ class QRadioButton;
...
+    void replaceAllSignal(QString, QTextDocument::FindFlags, QString);
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.h:31]: See comment above

File: wgtfindreplacetext.h [hunk 8]
```diff
@@ -38,10 +54,13 @@ class FindReplaceText : public QDialog
     QLabel *optionsLbl;
     QCheckBox *caseSensitiveBox;
     QCheckBox *wholeWordsBox;
-    QCheckBox *regularExprBox;
 
     //label status
     QLabel *statusLbl;
+
+    //data
+    QString newWord;
```
Comment (pmogylevskyi) [inline wgtfindreplacetext.h:62]: Style: consider to use m_ prefix for class variables

File: wgttextview.h [hunk 9]
```diff
@@ -39,13 +43,20 @@ class wgtTextView : public QMainWindow
     QPlainTextEdit *textView;
 
     QString pathToFile;
-    QString findText;
     FindReplaceText *findReplace;
+    void replace(QString, QString, QTextDocument::FindFlags);
```
Comment (pmogilevskiy) [inline wgttextview.h]: const QString &

File: wgttextview.h [hunk 10]
```diff
@@ -39,13 +43,20 @@ class wgtTextView : public QMainWindow
...
+    void replaceAll(QString, QString, QTextDocument::FindFlags);
 
 private slots:
     bool saveAs();
     bool saveFile();
-    bool findInText();
+
+    //slots for wgtfindreplacetext
+    void find();
+    void findSlot(QString, QTextDocument::FindFlags);
```
Comment (pmogylevskyi) [inline wgttextview.h:56]: See above

File: wgttextview.h [hunk 11]
```diff
@@ -39,13 +43,20 @@ class wgtTextView : public QMainWindow
...
+    void replaceSlot(QString, QTextDocument::FindFlags, QString);
```
Comment (pmogylevskyi) [inline wgttextview.h:57]: See above

File: wgttextview.h [hunk 12]
```diff
@@ -39,13 +43,20 @@ class wgtTextView : public QMainWindow
...
+    void replaceAllSlot(QString, QTextDocument::FindFlags, QString);
```
Comment (pmogylevskyi) [inline wgttextview.h:58]: See above

File: wgttextview.h (hunk 9 above)
Comment (pmogylevskyi) [inline wgttextview.h:47]: const QString &

File: wgttextview.cpp [hunk 13]
```diff
@@ -90,7 +131,20 @@ void wgtTextView::editFile(QString &filePath)
 
 void wgtTextView::closeEvent(QCloseEvent *)
 {
-    findReplace->close();
+    QMessageBox::StandardButton reply;
```
Comment (pmogylevskyi) [inline wgttextview.cpp:134]: Variable is not initialized

File: wgttextview.cpp [hunk 14]
```diff
@@ -90,7 +131,20 @@ void wgtTextView::editFile(QString &filePath)
...
+    if (textView->document()->isModified()) {
+        reply = QMessageBox::question(this, "Documet was modified",
```
Comment (pmogylevskyi) [inline wgttextview.cpp:136]: Missed tr("")

File: wgttextview.cpp [hunk 15]
```diff
@@ -90,7 +131,20 @@ void wgtTextView::editFile(QString &filePath)
...
+                                      "Do you want save?",
+                                      QMessageBox::Yes | QMessageBox::No);
+    }
+    if (reply == QMessageBox::Yes)
+    {
+        this->saveFile();
+        findReplace->close();
+    }
+    else {
+        findReplace->close();
```
Comment (pmogylevskyi) [inline wgttextview.cpp:146]: This line duplicates 143 line.

File: wgttextview.cpp [hunk 16]
```diff
@@ -7,17 +7,27 @@
...
+
+    //color theme
+    QPalette p = this->textView->palette();
+    p.setColor(QPalette::Base, Qt::white);
+    p.setColor(QPalette::Text, Qt::black);
+    p.setColor(QPalette::Highlight, Qt::green);
+    p.setColor(QPalette::HighlightedText, Qt::black);
+    this->textView->setPalette(p);
 
     findReplace = new FindReplaceText();
```
Comment (pmogylevskyi) [inline wgttextview.cpp:26]: Memory leak?

**Pull request merged**